FoodOrder.TestApi/Controller/RestaurantTestController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using AutoMapper;
+using FluentAssertions;
+using FoodOrderApi.Controllers;
+using FoodOrderApi.DataProvider;
+using FoodOrderApi.Mappings;
+using FoodOrderApi.Model.DTO;
+using FoodOrderApi.TestApi.Mockdata;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using System.Net;
+
+namespace FoodOrderApi.TestApi.Controller
+{
+    public class TestRestaurantController
+    {
+        private Mock<IDataProvider> restaurantService;
+        private IMapper mapperMock;
+        private Mock<ILogger<RestaurantController>> loggerMock;
+
+        public TestRestaurantController()
+        {
+            restaurantService = new Mock<IDataProvider>();
+            loggerMock = new Mock<ILogger<RestaurantController>>();
+            if (mapperMock == null)
+            {
+                var mappingConfig = new MapperConfiguration(mc =>
+                {
+                    mc.AddProfile(new AutoMapperProfiles());
+                });
+                IMapper mapper = mappingConfig.CreateMapper();
+                mapperMock = mapper;
+            }
+        }
+
+        [Theory]
+        [InlineData(1, 10)]
+        [InlineData(3, 2)]
+        public async Task Get_AllRestaurantPaginationWithValidIndex__ShouldReturn200Status(int pageNumber, int pageSize)
+        {
+            /// Arrange
+            restaurantService.Setup(_ => _.GetRestaurantPaged(pageNumber, pageSize)).ReturnsAsync(RestaurantMockData.GetRestaurantPaged(pageNumber, pageSize));
+            var sut = new RestaurantController(restaurantService.Object, mapperMock, loggerMock.Object);
+            sut.ControllerContext = new ControllerContext();
+            sut.ControllerContext.HttpContext = new DefaultHttpContext();
+
+            /// Act
+            var result = await sut.GetRestaurant(pageNumber, pageSize);
+            var model = (result as OkObjectResult).Value as List<DisplayRestaurantDTO>;
+            // /// Assert
+            model.Should().HaveCount(RestaurantMockData.GetRestaurantPaged(pageNumber, pageSize).Item1.Count());
+            result.Should().BeOfType<OkObjectResult>().Which.StatusCode.Should().Be((int)HttpStatusCode.OK);
+        }
+
+        [Theory]
+        [InlineData(11, 10)]
+        [InlineData(10, 7)]
+        public async Task Get_AllRestaurantPaginationWithInvalidIndex__ShouldReturn404Status(int pageNumber, int pageSize)
+        {
+            /// Arrange
+            restaurantService.Setup(_ => _.GetRestaurantPaged(pageNumber, pageSize)).ReturnsAsync(RestaurantMockData.GetRestaurantPaged(pageNumber, pageSize));
+            var sut = new RestaurantController(restaurantService.Object, mapperMock, loggerMock.Object);
+            sut.ControllerContext = new ControllerContext();
+            sut.ControllerContext.HttpContext = new DefaultHttpContext();
+
+            /// Act
```

Comment:
Remove the Comments that say Arrange, Act, Assert. Separate the Phases using a single line. Do the same for other locations as well.

Reply:
Removed the comments. Some places where we referred has given it with the comments, Is it okay without mentioning them?